internal/resolvers/youtube/model.go

```diff
@@ -9,3 +9,15 @@ type youtubeTooltipData struct {
 	LikeCount    string
 	DislikeCount string
 }
+
+type youtubeChannelTooltipData struct {
+	Title        string
+	PublishDate  string
+	Subscribers  string
+	Views        string
+}
+
+type ChannelId struct {
+	id string
+	channelType ChannelType
+}
```

Comment:
This struct seems off to me. If I understand it correctly, it's used to structure the cache key used when accessing the youtube channel cache.

You could move this to its own file, call it `ChannelCacheKey`, and have a method in that file for stringifying it, and a function for creating one from the string instead.

It also shouldn't be exported publicly (i.e. have an uppercase first character) if it's not being used in any other package.
The functions I'm referring to that should be part of that file instead are `constructCacheKeyFromChannelId` and `deconstructChannelIdFromCacheKey`

Reply:
The thinking was it's just easier to work with the channels types when we can infer the type from the key. Used to tuple returns in C# so did something similar. 

Moved it!